File: backend/app/application/dto/certification.py
"""Application DTO for public certifications."""

from dataclasses import dataclass
from datetime import date, datetime
from uuid import UUID

from app.domain.entities.certification import Certification


@dataclass(frozen=True, slots=True)
class CertificationDTO:
    """Framework-independent certification representation."""

    id: UUID
    name: str
    issuer: str
    credential_url: str | None
    issued_at: date
    expiration_date: date | None
    sort_order: int
    created_at: datetime
    updated_at: datetime

    @classmethod
    def from_entity(cls, certification: Certification) -> "CertificationDTO":
        return cls(
            id=certification.id,
            name=certification.name,
            issuer=certification.issuer,
            credential_url=certification.credential_url,
            issued_at=certification.issued_at,
            expiration_date=certification.expiration_date,
            sort_order=certification.sort_order,
            created_at=certification.created_at,
            updated_at=certification.updated_at,
        )

File: frontend/src/schemas/certification.ts
import { z } from "zod";

export const certificationSchema = z.object({
  id: z.uuid(),
  name: z.string(),
  issuer: z.string(),
  credential_url: z.url().nullable(),
  issued_at: z.iso.date(),
  expiration_date: z.iso.date().nullable(),
  sort_order: z.number().int(),
  created_at: z.iso.datetime(),
  updated_at: z.iso.datetime(),
});

export const certificationsSchema = z.array(certificationSchema);
